Runtime/Tracking/Collision/CollisionTracker.cs

```diff
@@ -64,7 +113,87 @@ protected virtual void OnTriggerExit(Collider collider)
                 return;
             }
 
+            RemoveTrackedNode(collider);
             OnCollisionStopped(eventData.Set(this, true, null, collider));
         }
+
+        /// <summary>
+        /// Adds a <see cref="CollisionTrackerDisabledObserver"/> to the <see cref="GameObject"/> that contains the <see cref="Collider"/> causing the collision.
+        /// </summary>
+        /// <param name="target">The target to add the <see cref="CollisionTrackerDisabledObserver"/> component to.</param>
+        protected virtual void AddTrackedNode(Collider target)
+        {
+            if (target == null || !StopCollisionsOnDisable)
+            {
+                return;
+            }
+
+            trackedCollisions.Add(target);
+            CollisionTrackerDisabledObserver node = target.gameObject.AddComponent<CollisionTrackerDisabledObserver>();
+            node.Source = this;
+            node.Target = target;
+        }
+
+        /// <summary>
+        /// Removes the <see cref="CollisionTrackerDisabledObserver"/> from the <see cref="GameObject"/> that contains the <see cref="Collider"/> ceasing the collision.
+        /// </summary>
+        /// <param name="target">The target to remove the <see cref="CollisionTrackerDisabledObserver"/> component from.</param>
+        protected virtual void RemoveTrackedNode(Collider target)
+        {
+            if (target == null || !StopCollisionsOnDisable)
+            {
+                return;
+            }
+
+            foreach (CollisionTrackerDisabledObserver node in target.gameObject.GetComponents<CollisionTrackerDisabledObserver>().ToArray())
+            {
+                if (node.Source == this && node.Target == target)
+                {
+                    node.Destroy();
+                    trackedCollisions.Remove(target);
+                }
+            }
+        }
+    }
+
+    /// <summary>
+    /// Observes the disabled state of any <see cref="GameObject"/> that the <see cref="CollisionTracker"/> is currently colliding with.
+    /// </summary>
+    public class CollisionTrackerDisabledObserver : MonoBehaviour

```

Comment:
Put in standalone file instead?

Reply:
Well if it's in a standalone file it can be dragged into the inspector, which we don't really want. It's basically the "magic" helper for this CollisionTracker.

Reply:
`[AddComponentMenu("")]` should hide it from there! I think we even use that elsewhere already :)

Reply:
It hides it from the drop down but doesn't prevent you from dragging that component into the inspector window, which we still dont want.

Reply:
🤷‍♂ There's only so much I'd do to prevent obvious mistakes.

I just don't like relying on this Unity behavior as it seems undocumented, or at least not documented as a "feature" that is intended this way..